core/exception/detailed_message_spec.rb

```diff
@@ -0,0 +1,35 @@
+require_relative '../../spec_helper'
+
+describe "Exception#detailed_message" do
+  ruby_version_is "3.2" do
+    it "returns decorated message" do
+      RuntimeError.new("new error").detailed_message.should == "new error (RuntimeError)"
+    end
+
+    it "accepts highlight keyword argument adn adds escape conrol sequences" do
+      RuntimeError.new("new error").detailed_message(highlight: true).should == "\e[1mnew error (\e[1;4mRuntimeError\e[m\e[1m)\e[m"
+    end
+
+    it "allows and ingores other keyword arguments" do
+      RuntimeError.new("new error").detailed_message(foo: true).should == "new error (RuntimeError)"
+    end
+
+    it "returns just a message if exception class is anonymous" do
+      Class.new(RuntimeError).new("message").detailed_message.should == "message"
+    end
+
+    it "returns 'unhandled exception' for an instance of RuntimeError with empty message" do
+      RuntimeError.new("").detailed_message.should == "unhandled exception"
+    end
+
+    it "returns just class name for an instance of Exception sublass with empty message" do
+      C = Class.new(Exception)
+      C.new("").detailed_message.should == "C"
+    end
```

Comment:
Is `Exception` used intentionally here?

I suggested to check a subclass of `RuntimeError` (not `Exception`) to emphasise that only `RuntimeError` behaves this way and subclasses don't inherit it.

It's OK to have this test case as is (as well as a test case below), but I would find useful to also check in a separate test case a `RuntimeError` subclass.

Reply:
Also defining new constants if prohibited and is checked on CI.

Some jobs fail with error message:

```
ConstantLeakError: Top level constants leaked: C
```

The recommended way to introduce new test-specific constants is to define them in `...Specs` helper modules.
There is a module for Exception specs in `core/exception/fixtures/common.rb`